### PR TITLE
chore: Reduce debug logs

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

When we compile the minimum version of node support, it will generate tens of thousands of lines of log, which is useless most of the time.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14806"></a>

### CODE REVIEW

**Comment by liuxingbaoyu** [comment]
Also why `env` is `development` in our babel8 ci, shouldn't it be `production`?

**Comment by nicolo-ribaudo** [comment]
We use `development` for unit tests, and `production` for e2e tests. This is because with `development` we have more debug info (and we have unbundled files so we can run more tests), while e2e tests must match the published version.

**Comment by nicolo-ribaudo** [comment]
> When we compile the minimum version of node support, it will generate tens of thousands of lines of log, which is useless most of the time.

I'm surprised, we should only log once anyway :thinking:

**Comment by liuxingbaoyu** [comment]
> We use `development` for unit tests, and `production` for e2e tests. This is because with `development` we have more debug info (and we have unbundled files so we can run more tests), while e2e tests must match the published version.

Oh, yes. Although `test-legacy` falls through to `production`, they are not actually the same.

> I'm surprised, we should only log once anyway 🤔

https://github.com/babel/babel/runs/7554396926?check_suite_focus=true
I also thought pollyfill would output one per use.😂

**Comment by JLHwung** [comment]
It seems to me the preset is loaded with cache miss, which should not happen if properly configured. Can you reproduce this issue on Node < 16?

**Comment by nicolo-ribaudo** [comment]
I can reproduce this by forcing our build process to not use workers.

**Comment by liuxingbaoyu** [comment]
I remember this happening locally before, but I can't seem to reproduce it now. (will only print a few times instead of thousands, I don't know if that's expected)

**Comment by nicolo-ribaudo** [comment]
More likely, it's a race condition in how we use gensync than a bug in gensync itself.

**Comment by liuxingbaoyu** [comment]
> https://github.com/babel/babel/blob/e2442e15cae4b3532adcc7a61a8cf1d7f7d23bf6/babel-worker.cjs#L35
> 
> I can workaround the issue by using `transformSync` here, which suggests that the caching issue might be a `gensync` bug. Can you update the PR?

This is awesome, thanks!

I've tried debugging, but the methods are way too complicated and totally clueless. (they are wrapped by gensync)😭

File: babel.config.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ module.exports = function (api) {
       break;
     case "development":
       envOpts.debug = true;
+
```
Comment (JLHwung) [inline babel.config.js:135]:
```suggestion
```

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
This is good for now, but I eventually want to go back to the async transform (for ESM).

**Comment by JLHwung** [comment]
> go back to the async transform (for ESM).

Yes, I can reproduce the issue in `packages/babel-core/test/config-loading.js`

```js
it("should load and cache the config with plugins and presets", async () => {
      const opts = makeOpts();

      const [options1, options2] = await Promise.all([
        loadOptionsAsync(opts),
        loadOptionsAsync(opts),
      ]);
      expect(options1.plugins.map(p => p.key)).toEqual([
        "plugin1",
        "plugin2",
        "plugin6",
        "plugin5",
        "plugin4",
        "plugin3",
      ]);

      expect(options2.plugins.length).toBe(options1.plugins.length);

      for (let i = 0; i < options2.plugins.length; i++) {
        expect(options2.plugins[i]).toBe(options1.plugins[i]);
      }
    });
```
Currently the last assertions are failing. `options1.plugins[i]` and `options2.plugins[i]` can be serialized to the same string but they are not the same object, so every plugins or presets are loaded without cache. If I removed `promise.all`, the issue can't be reproduced. So it is very likely due to racing conditions in the codebase.

**Pull request merged**